src/database.ts

```diff
@@ -832,36 +980,46 @@ class Database extends ServiceObject {
    *   const sessions = data[0];
    * });
    */
-  getSessions(options, callback) {
+  getSessions(
+    optionsOrCallback?: GetSessionsOptions | GetSessionsCallback,
+    cb?: GetSessionsCallback
+  ): void | Promise<GetSessionsResponse> {
     const self = this;
-    if (is.fn(options)) {
-      callback = options;
-      options = {};
-    }
-    const gaxOpts = options.gaxOptions;
-    const reqOpts = extend({}, options, {database: this.formattedName_});
+    const callback =
+      typeof optionsOrCallback === 'function'
+        ? (optionsOrCallback as GetSessionsCallback)
+        : cb;
+    const options =
+      typeof optionsOrCallback === 'object'
+        ? (optionsOrCallback as GetSessionsOptions)
+        : {gaxOptions: {}};
+    const gaxOpts: CallOptions = options.gaxOptions as CallOptions;
+    const reqOpts = extend({}, options, {
+      database: this.formattedName_,
+    });
+
     delete reqOpts.gaxOptions;
-    this.request(
+    this.request<Session, google.spanner.v1.IListSessionsResponse>(
```

Comment:
Pretty sure this is wrong, it looks like the first argument is `ISession` and that we map it to a `Session` object.